src/Splitter.hs

```diff
@@ -1,56 +1,310 @@
 {-# LANGUAGE RecordWildCards #-}
+{-# Language QuasiQuotes #-}
+{-# Language GADTs #-}
+{-# Language OverloadedStrings #-}
+{-# Language LambdaCase #-}
 module Splitter where
 
 import Syntax
 import RefinedAst
-import EVM (VM)
+import ErrM
+import Data.Text (Text, pack, unpack)
+import Data.List
+import qualified Data.List.NonEmpty as NonEmpty
+import Data.Maybe
+import Data.ByteString hiding (pack, unpack, intercalate, foldr, concat, head, tail)
+import qualified Data.Text as Text
+import Parse
+import Data.Bifunctor
+import EVM (VM, ContractCode)
+import EVM.Types
 import EVM.Symbolic (verify, Precondition, Postcondition)
---import EVM.Solidity (AbiType)
-import EVM.Solidity (SolcContract)
-import Data.Map.Strict    (Map) -- abandon in favor of [(a,b)]?
-
--- data Implementation = Implementation
---   {_contractBinaries :: Map Id Code,
---    _storageLayout :: Map Id Layout
---   }
-
--- type Code = (String, String)
-
--- data Layout = Layout {
---   slot :: Int,
---   offset :: Int,
---   label :: Id,
---   encoding :: Encoding,
---   length :: Int
---   }
--- data Encoding = Inplace | Mapping | DynamicArray | Bytes
--- data KSpec = KSpec {
---   k :: KTerm,
---   program :: String,
---   jumpDests :: String,
---   callData :: String,
---   output :: ReturnExp,
---   statusCode :: StatusCode,
---   kreturn :: String
---  accounts :: Map Contract
--- deriving (Ord, Eq, Read)
-
--- instance Show KSpec where
---   show KSpec { .. }  = error "TODO"
-
--- data KTerm = Execute | Halt
---   deriving (Ord, Eq, Read)
-
--- instance Show KTerm where
---   show Execute = "execute"
---   show Halt = "halt"
-
--- data StatusCode
---   = EVMC_SUCCESS
---   | EVMC_REVERT
---   | EVMC_OOG
---   deriving (Show, Ord, Eq, Read)
---type Signature = (Text, AbiType)
-
--- kaseSplit :: Behaviour -> SolcContract -> (VM, Signature, Precondition, Postcondition)
--- kaseSplit = error "TODO"
+
+import EVM.Solidity (SolcContract(..), StorageItem(..), SlotType(..))
+import Control.Monad
+import Data.Map.Strict (Map) -- abandon in favor of [(a,b)]?
+import qualified Data.Map.Strict as Map -- abandon in favor of [(a,b)]?
+
+-- Transforms a RefinedSyntax.Behaviour
+-- to a k spec.
```

Comment:
Maybe the file name could represent that better?

Reply:
Yeah, the file name should change. I was afraid git would screw up the diff if I did this in the same PR. Will go in the next one!

Reply:
Ok!